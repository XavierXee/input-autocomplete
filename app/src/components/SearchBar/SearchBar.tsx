import React from 'react';
import api from './../../utils/api';

import {AutoCompletionEntry} from "../../interfaces/AutoCompletionEntry";

import AutoCompleteInput from "./../AutoCompleteInput/AutoCompleteInput";
import AutoCompleteSubmit from "./../AutoCompleteSubmit/AutoCompleteSubmit";
import AutoCompleteEntries from "./../AutoCompleteEntries/AutoCompleteEntries";

import './SearchBar.scss';
import {AutoCompleteEntriesProps} from "../../interfaces/AutoCompleteEntriesProps";

class SearchBar extends React.Component<{}, { searchString: string, errorMessage: string, autoCompletionData: any[]}> {
// class SearchBar extends React.Component<{}, { searchString: string, errorMessage: string, autoCompletionData: AutoCompletionEntry[]}> {
    constructor(props: any) {
        super(props);
        this.state = {
            searchString: '',
            errorMessage: '',
            autoCompletionData: []
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleBlur = this.handleBlur.bind(this);
    }

    // TODO: type event
    handleChange(value: any): void {
        this.setState({searchString: value});
        if (value) {
            api.getAutoCompletion(value)
                .then((result) => {
                    this.setState({autoCompletionData: result});
                    console.log(this.state);
                })
                .catch(() => {

                })
        } else {
            this.setState({autoCompletionData: []});
        }
    }

    handleSubmit(): void {
        alert('Le nom a été soumis : ' + this.state.searchString);
    }

    handleBlur(): void {
        this.setState({
            searchString: '',
            autoCompletionData: []
        });
    }

    render() {
        return (
            <div className="SearchBar">
                <form className="SearchBar--form">
                    <AutoCompleteInput searchString={this.state.searchString} onBlur={this.handleBlur} onChange={this.handleChange}/>
                    <AutoCompleteSubmit onSubmit={this.handleSubmit}/>
                    <AutoCompleteEntries autoCompletionData={this.state.autoCompletionData}/>
                </form>
            </div>
        );
    }
}

export default SearchBar;
