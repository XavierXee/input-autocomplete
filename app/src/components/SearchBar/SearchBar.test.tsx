import React from 'react';
import ReactDOM from 'react-dom';
import SearchBar from './SearchBar';
import { create } from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SearchBar />, div);
});

it('render correctly', () => {
    const component = create(<SearchBar />).toJSON();;
    expect(component).toMatchSnapshot();
});