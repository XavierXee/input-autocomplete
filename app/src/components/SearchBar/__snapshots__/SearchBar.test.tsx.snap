// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="SearchBar"
>
  <form
    className="SearchBar--form"
  >
    <span
      className="AutoCompleteInput"
    >
      <i
        className="AutoCompleteInput--icon fas fa-search"
      >
        Â 
      </i>
      <input
        className="AutoCompleteInput--field"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
    </span>
    <button
      className="AutoCompleteSubmit"
      onClick={[Function]}
    >
      Search
    </button>
    <div
      className="AutoCompleteEntries"
    >
      <ul
        className="AutoCompleteEntries--list"
      >
        
      </ul>
    </div>
    
  </form>
</div>
`;
